@xen-orchestra/lite/src/components/RangeInput.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import { withState } from 'reaclette'
+
+type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>
+
+interface InputAttributes extends React.AllHTMLAttributes<HTMLInputElement> {
+  max: number
+  min: number
+  step: number
+}
+
+interface ParentState {}
+
+interface State {
+  currentValue: number
+  input: React.RefObject<HTMLInputElement>
+}
+
+interface Props {
+  defaultValue: number
+  inputAttribues: Omit<InputAttributes, 'onChange'>
+  onChange: (value: number) => void
+}
+
+interface ParentEffects {}
+
+interface Effects {
+  _onChange: React.ChangeEventHandler<HTMLInputElement>
+}
+
+interface Computed {}
+
+const RangeInput = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: props => ({
+      currentValue: props.defaultValue,
+      input: React.createRef(),
+    }),
+    effects: {
+      _onChange: function (e) {
+        const { value } = e.currentTarget
+
+        this.state.currentValue = +value
+        this.props.onChange(+value)
+      },
+    },
+  },
+  ({ effects, inputAttribues, state }) => (
```

Comment:
I created an extra prop, because by using the rest operator to get all the props, am getting some props that i dont want.

Some exemples:
i get the `onChange` prop, and it will overide the `onChange` value of the input.
I get the `resetState` prop, and he will try to inject this function into the `input` tag, and generate an error in the web console.

By creating extra prop, this give me more control of what i want inject into the `input` tag and let me decide if i want to omit some `HTML input function` like what i did for `onChange` [here](https://github.com/vatesfr/xen-orchestra/pull/5703/files#diff-8365d3997b0fda0405f43030763b8247b7d07963364eed4bb039ea4d22fefa9eR21).

Reply:
Ok, but when you're using this component, you shouldn't have to know which props go into the `inputAttributes` and which ones need to be passed directly.

If you want to grab some props for your implementation, you can destructure them before the rest operator and then forward everything else to the input:
```js
({ effects, state, resetState, onChange, ...props }) =>
```